spec.bs

```diff
@@ -958,6 +959,125 @@ For now, see [https://github.com/wicg/observable#operators](https://github.com/w
+          algorithm's <a href=#switchmap-next-steps>next steps</a> (below), and only assigned to
+          null by the [=switchmap process next value steps=], when the "inner" {{Observable}} either
+          completes or errors. This variable is used as a marker for whether there is currently an
+          active "inner" subscription. The <a href=#switchmap-complete-steps>complete steps</a>
+          below care about this, because if |sourceObservable| completes while there is an active
+          "inner" subscription, we do not immediately complete |subscriber|. In that case,
+          |subscriber|'s completion becomes blocked on the "inner" subscription's completion.
+
+       1. Let |sourceObserver| be a new [=internal observer=], initialized as follows:
+
+          : <dt id=switchmap-next-steps>[=internal observer/next steps=]</dt>
+          :: 1. If |activeInnerAbortController| is not null, then [=AbortController/signal abort=]
+                |activeInnerAbortController|.
+
+                Note: This "unsubscribes" from the "inner" {{Observable}} that was derived from the
+                value that was <i>last</i> pushed from |sourceObservable|. Then we immediately
+                subscribe to the <i>new</i> {{Observable}} that we're about to derive from |value|,
+                i.e., the <i>most-recently</i> pushed value from |sourceObservable|.
+
+             1. Set |activeInnerAbortController| to a [=new=] {{AbortController}}.
+
+             1. Run the [=switchmap process next value steps=] with |value|, |subscriber|, |mapper|,
+                and <b>references</b> to all of the following: |activeInnerAbortController|, and
+                |outerSubscriptionHasCompleted|.
+
+                Note: The [=switchmap process next value steps=] will subscribe to the
+                {{Observable}} derived from |value| (if one such can be derived) and keep processing
+                values from it until either (1) its subscription becomes inactive (either by error or
+                completion), or (2) |activeInnerAbortController| <a href=#switchmap-next-steps>gets
+                aborted</a>, due to |sourceObservable| pushing another <i>newer</i> value that will
+                replace the current "inner" subscription.
+
+          : [=internal observer/error steps=]
+          :: Run |subscriber|'s {{Subscriber/error()}} method, given the passed in <var
+             ignore>error</var>.
+
+          : <dt id=switchmap-complete-steps>[=internal observer/complete steps=]</dt>
+          :: 1. Set |outerSubscriptionHasCompleted| to true.
+
+                Note: If |activeInnerAbortController| is not null, then we don't immediately
+                complete |subscriber|. Instead, the [=switchmap process next value steps=] will
+                complete |subscriber| when the inner subscription finally becomes inactive on its
+                own volition.
```

Comment:
It will only complete if the inner subscriber completes. "inactive" almost implies that if it waits a while or maybe errors, it will still complete. Both the outer source and the inner source, if there is one, have to complete in order for the subscriber to complete.

Reply:
Improved this a bit. Done.